### PR TITLE
Use priority class `gardener-system-critical`

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:
This PR removes priority class `gardener-system-critical-migration` introduced in https://github.com/gardener/gardener/pull/6510 and uses `gardener-system-critical` priority class for all the components which are supposed to use `gardener-system-critical` priority class as described in https://github.com/gardener/gardener/issues/5634.

**Which issue(s) this PR fixes**:
Fixes Part of https://github.com/gardener/gardener/issues/5634

**Special notes for your reviewer**:
/cc @ialidzhikov 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by shafeeqes — COMMENTED** [review]
Thanks!
/lgtm

File: pkg/gardenlet/controller/factory.go [hunk 1]
```diff
@@ -135,7 +135,8 @@ func (f *GardenletControllerFactory) Run(ctx context.Context) error {
 		return fmt.Errorf("failed to get seed client: %w", err)
 	}
 
-	if err := client.IgnoreNotFound(seedClient.Client().Delete(ctx, &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "gardenlet"}})); err != nil {
+	// TODO: Remove in a future release.
```
Comment (ialidzhikov) [inline pkg/gardenlet/controller/factory.go]:
```suggestion
	// TODO(acumino): Remove in a future release.
```

**Review by ialidzhikov — COMMENTED** [review]
Nit otherwise lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/approve

**Pull request merged**